apps/server/src/shared/domain/service/lesson-copy.service.ts

```diff
@@ -19,22 +19,71 @@ export class LessonCopyService {
 			hidden: true,
 			name: this.nameCopyService.deriveCopyName(params.originalLesson.name),
 			position: params.originalLesson.position,
+			contents: this.copyTextComponent(params.originalLesson),
 		});
 
+		const elements = [...this.lessonStatusElements(), ...this.textComponentStatus(copy.contents)];
+
 		const status: CopyStatus = {
 			title: copy.name,
 			type: CopyElementType.LESSON,
-			status: CopyStatusEnum.SUCCESS,
+			status: CopyStatusEnum.PARTIAL,
 			copyEntity: copy,
-			elements: [
-				{
-					title: 'metadata',
-					type: CopyElementType.LEAF,
-					status: CopyStatusEnum.SUCCESS,
-				},
-			],
+			elements,
 		};
 
 		return status;
 	}
+
+	private copyTextComponent(originalLesson: Lesson): IComponentProperties[] {
+		const copiedContent: IComponentProperties[] = [];
+		const textComponents = originalLesson.contents.filter((element) => Object.keys(element.content)[0] === 'text');
```

Comment:
note for latter: we're gonna have to see how can we keep the position of elements inside the array
something like copyComponents, and for each type apply a copy logic